src/Cms/Auth.php

```diff
@@ -712,9 +712,13 @@ public function track(string|null $email, bool $triggerHook = true): bool
 	public function type(bool $allowImpersonation = true): string
 	{
 		$basicAuth = $this->kirby->option('api.basicAuth', false);
-		$auth      = $this->kirby->request()->auth();
+		$request   = $this->kirby->request();
 
-		if ($basicAuth === true && $auth && $auth->type() === 'basic') {
+		if (
+			$basicAuth === true &&
+			$request->auth() &&
+			$request->auth()->type() === 'basic'
```

Comment:
Could be shortened to `$request->auth()?->type() === 'basic'`

Reply:
No, because `$request->auth()` can return `false`, which would lead to "call to type() on bool".

Reply:
👍

Reply:
But can I ask what this change is for? It seems to have the same functionality as before, just now spelling out `$request->auth()` twice instead keeping it in a variable

Reply:
See the unit tests. If we call `auth()` beforehand, it will always trigger `$app->response()->usesAuth()`, even if the auth is never actually used.

Reply:
Maybe we can add a comment for us others to later still understand that basically `$basicAuth === true` is the condition that really needs to be true before we even call `->auth()`. I'm afraid we'll read it in the future and this won\t be so clear to us anymore.